src/azul/chalice.py

```diff
@@ -504,34 +498,109 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    @attr.define(frozen=True, kw_only=True)
+    class _lambda_config:
+        lambda_name: Optional[str] = attr.ib(default=None)
+        handler_name: Optional[str] = attr.ib(default=None)
+
+        def bind(self, app_name: str, handler_name: Optional[str] = None, /) -> Self:
+            lambda_name, _ = config.unqualified_resource_name(app_name)
+
+            return attr.evolve(self,
+                               lambda_name=lambda_name,
+                               handler_name=handler_name)
+
+        @property
+        def lambda_tf_resource_name(self) -> str:
+            if self.handler_name is None:
+                return self.lambda_name
+            else:
+                assert self.handler_name != ''
+                return f'{self.lambda_name}_{self.handler_name}'
+
+    # noinspection PyPep8Naming
+    @attr.define(frozen=True)
```

Comment:
```suggestion
    @attr.define(frozen=True, kw_only=True)
```

The other subclass is fine as-is since it only has one non-default attribute.